app/utils/metrics.js

```diff
@@ -1,13 +1,45 @@
 'use strict'
 
+const net = require('net')
+const logger = require('winston')
+const certinfo = require('cert-info')
 const appmetrics = require('appmetrics')
 const metricsHost = process.env.METRICS_HOST || 'localhost'
 const metricsPort = process.env.METRICS_PORT || 8125
 const metricsPrefix = 'frontend.'
+const { APPLE_PAY_MERCHANT_ID_CERTIFICATE } = process.env
 
-exports.metrics = () => {
+const sendInitialCertificateMetric = function sendInitialCertificateMetric () {
+  if (APPLE_PAY_MERCHANT_ID_CERTIFICATE !== undefined) {
+    var merchantIdCert = certinfo.info(APPLE_PAY_MERCHANT_ID_CERTIFICATE)
+    var certificateTimeToExpiry = merchantIdCert.expiresAt - Date.now()
+    logger.debug(`Merchant identity cert will expire in ${certificateTimeToExpiry} milliseconds`)
+    const certificateKey = 'apple-pay.merchant-identity-certificate.time-to-expiry'
+    try {
+      const socket = net.createConnection(metricsPort, metricsHost, () => {
+        logger.debug(`Socket opened with ${metricsHost} ${metricsPort}`)
+        socket.write(`${metricsPrefix}.${certificateKey} ${certificateTimeToExpiry}\n`)
+        socket.end()
+      })
+
+      socket.on('data', (data) => {
+        logger.debug('Socket said', data)
```

Comment:
If this will be kept in production code it should say give more information. Maybe something along the lines of 'Data event from certificate expiration socket:'